include/tvm/relay/op_attr_types.h

```diff
@@ -185,6 +185,20 @@ using FTVMLegalize = runtime::TypedPackedFunc<Expr(const Attrs& attrs, const Arr
 using FTVMAnnotateTarget = runtime::TypedPackedFunc<bool(const Attrs& attrs,  // NOLINT(*)
                                                          const Array<Expr>& args)>;
 
+/*!
+ * \brief Annotates an expression to indicate if an op should be compiled using
+ *  the given compiler/target.
+ *  \param attrs The attribute of the original expr.
+ *  \param args The arguments of the original expr.
+ *  \param out_type The return type of the original expr.
+ *
+ *  \return true if this op should be registered to invoke a specific compiler
+ *  for codegen, otherwise, false.
+ */
+using FTVMAnnotateTarget3 =
```

Comment:
The goal was to minimize changes to other existing runtimes (which is obviously not possible in case of `FTVMAnnotateTarget = runtime::TypedPackedFunc<bool(const Expr&)>`). Otherwise it looks to me as separate PR for this generalisation. What would be your advice here?

Reply:
I can totally understand your point about minimizing the changes, but I personally don't think it's a big issue given that we only have a few existing codegen. Since BYOC is getting mature, we should avoid ad hoc implementation as we can. In fact, we are planning to do another refactoring to cleanup redundant code.

The changing (or addition) of annotate target, which is one of the most important interfaces to codegen developers, are better to start with an RFC. It is fine even the RFC is small and simple. The main purposes are 1) making sure this change is compatible to all cases, and 2) letting everyone know you are going to make this change.

Reply:
extracted to [#6786](https://github.com/apache/incubator-tvm/pull/6786)